hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,406 @@
+        status=1
+    fi
+
+    ip netns del "${reset_interface_netns}"
+
+    ip link set "${interface}" up
+
+    return ${status}
+}
+
+get_pcis_from_pfs(){
+    local worker_netns="$1"
+    shift
+    local interfaces="$@"
+    for interface in $interfaces; do
+        pcis["$interface"]="$(get_pci_from_net_name "$interface" "$worker_netns")"
+    done
+}
+
+get_pci_from_net_name(){
+    local interface_name=$1
+    local worker_netns="$2"
+
+    if [[ -z "$(ip l show $interface_name)" ]];then
+        if [[ -n "$(docker exec -t ${worker_netns} ip l show $interface_name)" ]];then
+            ip netns exec ${worker_netns} bash -c "basename \$(readlink /sys/class/net/${interface_name}/device)"
+            return 0
+        fi
+        echo ""
+        return 1
+    fi
+
+    basename $(readlink /sys/class/net/${interface_name}/device)
+    return 0
+}
+
+netns_create(){
+    local worker_netns="$1"
+
+    if [[ ! -e /var/run/netns/$worker_netns ]];then
+        local pid="$(docker inspect -f '{{.State.Pid}}' $worker_netns)"
+
+        if [[ -z "$pid" ]];then
+            return 1
+        fi
+
+        mkdir -p /var/run/netns/
+        rm -rf /var/run/netns/$worker_netns
+        ln -sf /proc/$pid/ns/net "/var/run/netns/$worker_netns"
+
+        if [[ -z "$(ip netns | grep $worker_netns)" ]];then
+            return 1
+        fi
+    fi
+    return 0
+}
+
+switch_pfs(){
+    local worker_netns="$1"
+    shift
+    local interfaces="$@"
+
+    echo "Switching \"$interfaces\" into $worker_netns ..."
+
+    for pf in $interfaces;do
+        switch_pf "$pf" "$worker_netns"
+    done
+}
+
+switch_pf(){
+    local pf_name="$1"
+    local worker_netns="$2"
+
+    if [[ -z "$(ip netns | grep ${worker_netns})" ]];then
+        echo "Error: Namespace $worker_netns not found!"
+        return 1
+    fi
+
+    if [[ -z "$(ip l show ${pf_name})" ]];then
+        if [[ -z "$(docker exec -t ${worker_netns} ip l show ${pf_name})" ]];then
+            echo "Error: Interface $pf_name not found..."
+            return 1
+        fi
+
+        echo "PF ${pf_name} already in namespace $worker_netns!"
+    else
+        if ! ip l set dev $pf_name netns $worker_netns;then
+            echo "Error: unable to set $pf_name namespace to $worker_netns!"
+            return 1
+        fi
+    fi
+
+    if ! docker exec -t ${worker_netns} ip l set $pf_name up;then
+        echo "Error: unable to set $pf_name to up!"
+        return 1
+    fi
+    
+}
+
+switch_vf(){
+    local vf_name="$1"
+    local worker_netns="$2"
+
+    if [[ -z "$(ip l show $vf_name)" ]];then
+        return 1
+    fi
+
+    if ip link set "$vf_name" netns "$worker_netns"; then
+      if timeout "$TIMEOUT"s bash -c "until ip netns exec $worker_netns ip link show $vf_name > /dev/null; do sleep $POLL_INTERVAL; done"; then
+          return 0
+      else
+          return 1
+      fi
+    fi
```

Comment:
Same as above. Should it Return 0?

Reply:
I think it should actually be returning 1 if it reached this point, added an else case to the outer if that would return 1, this way we will never reach outside the ifs.